barretenberg/cpp/src/barretenberg/flavor/flavor.hpp

```diff
@@ -379,6 +379,9 @@ MegaZKRecursiveFlavor_<UltraCircuitBuilder>>;
 template <typename T>
 concept HasDataBus = IsMegaFlavor<T>;
 
+template <typename T>
+concept DoesRecursiveIPA = IsAnyOf<T, UltraFlavor, UltraFlavorWithZK>;
```

Comment:
not ultra keccak because that will only be called on the root rollup circuit which will just verify the IPA claim.

Reply:
its a flavor that will attach an IPA proof and claim to its proof.